src/components/controller/units/UnitsTable.vue

```diff
@@ -63,16 +68,39 @@ onMounted(() => {
   hideOpenUnitPopupsTooltip.value = getPreference('hideOpenUnitPopupsTooltip', loginStore.username)
 })
 
-async function openUnit(unitId: string) {
+async function openUnit(unit: Unit, versionCheck = true) {
   error.value.openUnit = ''
+  currentUnit.value = unit
+  loadingOpenUnit.value = true
 
   try {
-    await unitsStore.checkUnitToken(unitId)
-    const routeData = router.resolve({ path: `/controller/manage/${unitId}/dashboard` })
-    window.open(routeData.href, '_blank')
+    await unitsStore.getUnitInfo(currentUnit.value.id)
+    await unitsStore.getUnits()
+    // Find the now updated unit, as the currentUnit might have changed
+    currentUnit.value = unitsStore.units.find((u) => u.id == unit.id)!
+    // Logic on which message is shown is inside the ObsoleteApiModal component
+    let version = currentUnit.value?.info.api_version
+    if (version == '') {
+      version = '0.0.0'
+    }
+    if (versionCheck && !satisfies(version, REQUIRED_API_VERSION)) {
+      if (outside(version, REQUIRED_API_VERSION, '>')) {
+        showGreaterApiModal.value = true
```

Comment:
I guess `showObsoleteApiModal.value = false` is missing here. Maybe the best solution is to set:
```
showGreaterApiModal.value = false
showObsoleteApiModal.value = false
```
at the start of this function, i.e. at line 75

Reply:
Now that i think about it, the setting to false of the two flags is not needed, the only flag needed to be set to false is `showGreaterApiModal` since it's possible that is set to true due to the warning modal. Your call boss!